packages/core/core/src/types.ts

```diff
@@ -111,23 +111,44 @@ export interface UseCategory
   loading: ComputedProperty<boolean>;
 }
 
+export function AddToCart<PRODUCT>(product: PRODUCT, quantity: number): Promise<void>;
+export function AddToCart<PRODUCT>(product: PRODUCT, quantity: number, customQuery: {}): Promise<void>
+export function AddToCart<PRODUCT>(product: PRODUCT, quantity?: number, customQuery?: {}): any {
+  return { product, quantity, customQuery };
+}
+export function RemoveFromCart<CART_ITEM>(product: CART_ITEM): Promise<void>
+export function RemoveFromCart<CART_ITEM>(product: CART_ITEM, customQuery: {}): Promise<void>
+export function RemoveFromCart<CART_ITEM>(product: CART_ITEM, customQuery?: {}): any {
+  return { product, customQuery };
+}
+export function ApplyCoupon(coupon): Promise<void>
+export function ApplyCoupon(coupon, customQuery: {}): Promise<void>
+export function ApplyCoupon(coupon, customQuery?: {}): any {
+  return { coupon, customQuery };
+}
+export function LoadCart(customQuery?: {}): Promise<void>
+export function LoadCart(customQuery?: {}): any {
```

Comment:
@filrak look at this, he solved the problem with overloading functions using TS, actually I really like this - this case, if some platform doesn't have a custom query, you can easily handle this - we had similar problems in the past, remember? and we actually postponed solving that.

@lukasborawski can you create and issue to refactor other core interfaces in this way?
Let's say that goal could be - move each TS function definition under the composable interfaces to the separated function like you did. I'm pretty sure we'll find other places where overloading would be usefull

Reply:
@andrzejewsky I believe that @filrak already knows that. We spoke about this feature before.